File: csharp/sum-of-multiples/SumOfMultiples.cs
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max) =>
        (from m in multiples
         from n in Enumerable.Range(0, max)
         where n % m == 0
         select (n)).Distinct().Sum();
}

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*divs)
    @divs = divs
  end

  def to(stop)
    @divs.each_with_object([]) do |div, multiples|
      multiples.push *(div...stop).step(div)
    end.uniq.sum
  end
end
